src/vim9compile.c

```diff
@@ -1613,7 +1613,16 @@ compile_lhs(
 	    {
 		if (is_decl)
 		{
-		    semsg(_(e_variable_already_declared_str), lhs->lhs_name);
+		    // if we come here with what looks like an assignment like .=
+		    // but which has been reject by assignment_len() from may_compile_assignment
+		    // give a better error message
+		    char_u *p = skipwhite(lhs->lhs_end);
+		    if (p[0] == '.' && p[1] == '=')
+			emsg(_(e_dot_equal_not_supported_with_script_version_two));
+		    else if (!ends_excmd2(lhs->lhs_end, p))
```

Comment:
I think this may be wrong. I think it will cause to no longer cause the `e_variable_already_declared_str` message again. So I'll remove it again.